src/dtemplate.d

```diff
@@ -4105,6 +4109,44 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
                 result = match2;
                 return;
             }
+
+            /* Deduce common type
```

Comment:
Ok, can it be moved into a function in dcast?  I don't like the fact that it's using `impcnvResult` directly.